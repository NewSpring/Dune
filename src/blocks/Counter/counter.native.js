//@flow
import { Text, View, Platform, Button } from "react-native";

import type { ICounter } from "./";
import { counterStyle } from "./";

export const Counter = ({
  increment,
  counter,
  decrement,
  loading,
  sample,
}: ICounter) => (
  <View style={counterStyle.resolve()}>
    {loading && <Text>loading...</Text>}
    {sample &&
      <Text>
        GraphQL returned {sample.message} with a response of {sample.code}
      </Text>}
    <Text>Counter is at {counter} on {Platform.OS}</Text>
    <Button onPress={increment} title="Increment" />
    <Button onPress={decrement} title="Decrement" />
  </View>
);
Counter.displayName = "Counter";

export default Counter;
