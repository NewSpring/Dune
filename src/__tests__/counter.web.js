import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

import Counter from "../counter.web.js";

it("shows a counter count on web", () => {
  const tree = renderer.create(<Counter counter={1} />);
  expect(tree).toMatchSnapshot();
});

it("loading state", () => {
  const tree = renderer.create(<Counter counter={1} loading={true} />);
  expect(tree).toMatchSnapshot();
});

it("shows data from the server", () => {
  const sample = { code: 200, message: "hello world" };
  const tree = renderer.create(<Counter counter={1} sample={sample} />);
  expect(tree).toMatchSnapshot();
});

it("handles a click for increment", () => {
  const increment = jest.fn();
  const wrapper = shallow(<Counter counter={1} increment={increment} />);
  wrapper.find("button").first().simulate("click");
  expect(increment).toBeCalled();
});

it("handles a click for decrement", () => {
  const decrement = jest.fn();
  const wrapper = shallow(<Counter counter={1} decrement={decrement} />);
  wrapper.find("button").at(1).simulate("click");
  expect(decrement).toBeCalled();
});
