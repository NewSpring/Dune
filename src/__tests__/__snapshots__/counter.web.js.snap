// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loading state 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <h4>
    loading...
  </h4>
  <span>
    Counter is at 
    1
     on web
  </span>
  <br />
  <button
    onClick={undefined}
  >
    Increment
  </button>
  <button
    onClick={undefined}
  >
    Decrement
  </button>
</div>
`;

exports[`shows a counter count on web 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <span>
    Counter is at 
    1
     on web
  </span>
  <br />
  <button
    onClick={undefined}
  >
    Increment
  </button>
  <button
    onClick={undefined}
  >
    Decrement
  </button>
</div>
`;

exports[`shows data from the server 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <h4>
    GraphQL returned 
    hello world
     with a response of 
    200
  </h4>
  <span>
    Counter is at 
    1
     on web
  </span>
  <br />
  <button
    onClick={undefined}
  >
    Increment
  </button>
  <button
    onClick={undefined}
  >
    Decrement
  </button>
</div>
`;
